lib/ansible/modules/storage/netapp/sf_account_manager.py

```diff
@@ -0,0 +1,284 @@
+        type: CHAPSecret
+        description:
+        - CHAP secret to use for the initiator. Should be 12-16 characters long and impenetrable.
+        - The CHAP initiator secrets must be unique and cannot be the same as the target CHAP secret.
+        - If not specified, a random secret is created.
+
+    target_secret:
+        required: false
+        type: CHAPSecret
+        description:
+        - CHAP secret to use for the target (mutual CHAP authentication).
+        - Should be 12-16 characters long and impenetrable.
+        - The CHAP target secrets must be unique and cannot be the same as the initiator CHAP secret.
+        - If not specified, a random secret is created.
+
+    attributes:
+        required: false
+        type: dict
+        description: List of Name/Value pairs in JSON object format.
+
+    account_id:
+        required: false
+        note: required when action == 'delete' or action == 'update'
+        description:
+        - The ID of the account to manage or update
+
+    status:
+        required: false
+        type: str
+        description:
+        - Status of the account
+
+'''
+
+EXAMPLES = """
+    - name: Create Account
+      sf_account_manager:
+       hostname: "{{ solidfire_hostname }}"
+       username: "{{ solidfire_username }}"
+       password: "{{ solidfire_password }}"
+       action: create
+       name: TenantA
+
+    - name: Modify Account
+     sf_account_manager:
+       hostname: "{{ solidfire_hostname }}"
+       username: "{{ solidfire_username }}"
+       password: "{{ solidfire_password }}"
+       action: update
+       account_id: 7
+       name: AnsibleUser-Renamed
+
+    - name: Delete Account
+     sf_account_manager:
+       hostname: "{{ solidfire_hostname }}"
+       username: "{{ solidfire_username }}"
+       password: "{{ solidfire_password }}"
+       action: delete
+       account_id: 7
+"""
+
+RETURN = """
+msg:
+    description: Successful creation
+    returned: success
+    type: string
+    sample: '{"changed": true}'
+
+msg:
+    description: Successful update
+    returned: success
+    type: string
+    sample: '{"changed": true}'
+
+msg:
+    description: Successful deletion
+    returned: success
+    type: string
+    sample: '{"changed": true}'
+
+"""
+
+'''
+    Todo:
+        Enable account deletion by Name. Currently only possible through account_id.
+
+        Before updating an account, check the previous (current) attributes to currently report
+        the 'changed' property
+
+'''
+import sys
+import json
+import logging
+from traceback import format_exc
+
+from ansible.module_utils.basic import *
+from ansible.module_utils.urls import *
+from ansible.module_utils.pycompat24 import get_exception
+
+import socket
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+logging.basicConfig(level=logging.ERROR)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+from solidfire.factory import ElementFactory
```

Comment:
Import custom packages in try/except and handled with fail_json() in main() e.g.::

    try:
        import foo
        HAS_LIB=True
    except:
        HAS_LIB=False